x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -249,17 +251,21 @@ private static Query translate(BinaryComparison bc, TranslatorHandler handler) {
                 isDateLiteralComparison = true;
             }
 
+            ZoneId zoneId = null;
+            if (bc.left().dataType() == DATETIME) {
+                zoneId = bc.zoneId();
```

Comment:
What would be the downside of always getting the zoneId and passing it to the range query ?

Reply:
I don't see a "downside" other than possible confusion when one sees the translated query.
the timezone is only used for processing ES date math expressions.